src/Microsoft.Health.Fhir.Store.Utils/CancelRequest.cs

```diff
@@ -11,9 +11,9 @@ public class CancelRequest
     {
         private long _value;
 
-        internal bool IsSet => Interlocked.Read(ref _value) == 1;
+        public bool IsSet => Interlocked.Read(ref _value) == 1;
 
-        internal void Set()
+        public void Set()
```

Comment:
This class looks like `AutoResetEvent`, can this be used instead?

Reply:
This is already checked in code. This PR just make methods public. I suggest to leave as-is and investigate later,